CONTRIBUTE.md

```diff
@@ -0,0 +1,156 @@
+# Architecture
+
+Galaxykit has a fairly simple structure:
+
+```
+(galaxykit) henderson@mars => tree .
+.
+├── CHANGES.md
+├── CONTRIBUTE.md
+├── galaxykit
+│   ├── client.py
+│   ├── collections.py
+│   ├── command.py
+│   ├── container_images.py
+│   ├── containers.py
+│   ├── containerutils.py
+│   ├── groups.py
+│   ├── namespaces.py
+│   ├── registries.py
+│   └── users.py
+├── LICENSE.md
+├── README.md
+├── setup.cfg
+└── setup.py
+```
+
+Let's work through these files individually:
+
+### CHANGES.md
+A changelog, tracking all the updates to the code.
+
+### CONTRIBUTE.md
+
+This file, with advice on how to effectively contribute to galaxykit.
+
+### galaxykit:
+
+The source directory, containing all the code that goes into galaxykit.
+
+#### client.py
+
+This file contains the GalaxyClient object, which is essentially a single authenticated context for making requests against an existing galaxy_ng instance. This file and `command.py` contain the primary two interfaces to interacting with galaxykit.
+
+#### collections.py
+
+Functions for managing collections.
+
+#### command.py
+
+The command-line interface logic - this is what gets run when you invoke galaxykit directly on the command-line. Currently in a bit of a hacky state, as it's grown faster than we can refactor it. An extensive refactor is in progress - check out [this PR](https://github.com/ansible/galaxykit/pull/30) if you want to review the changes.
+
+#### container_images.py
+
+For interacting with individual container images.
+
+#### containers.py
+
+Functions for dealing with containers as as discrete units (deleting, setting metadata.)
```

Comment:
Can we explain the different concerns of container_images.py and containers.py a bit more?

Reply:
Really the two files should be merged in my opinion - iirc container_images.py only has one function in it. But I didn't want to do that on this PR, which should really just be adding the contributor's guide.